src/System.Net.Primitives/src/System/Net/NetworkInformation/NetworkInformationException.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace System.Net.NetworkInformation
+{
+    /// <devdoc>
+    ///    <para>
+    ///       Provides NetworkInformation exceptions to the application.
+    ///    </para>
+    /// </devdoc>
+    public class NetworkInformationException : Win32Exception
```

Comment:
@CIPop, sorry, I wasn't being very clear: I'd made a few comments about Win32Exception in previous networking-related PRs on GitHub, not in this one.  The comment was simply asking whether Win32Exception is the right base class.  We've made Win32Exception work on Unix, essentially wrapping an errno value and doing the appropriate strerror string translations, but it's an unfortunately named type.